opts.go

```diff
@@ -13,8 +32,16 @@ func WithEndpoint(endpoint string) Opt {
 }
 
 // WithCredentials sets the client credentials used in each request.
-func WithCredentials(c Credentials) Opt {
-	return WithGRPCDialOption(grpc.WithPerRPCCredentials(creds{c}))
+func WithCredentials(cr Credentials) Opt {
+	return func(c *Client) {
+		if kc, ok := cr.(keyCreds); ok {
+			c.openAPI.auth = openapi.BasicAuth{
+				UserName: kc.keyID,
+				Password: kc.keySecret,
+			}
+		}
+		WithGRPCDialOption(grpc.WithPerRPCCredentials(creds{cr}))(c)
```

Comment:
You will need to return the `WithGRPCDialOption`
```suggestion
		return WithGRPCDialOption(grpc.WithPerRPCCredentials(creds{cr}))(c)
```

Reply:
Oh sorry saw just saw the little `(c)` at the end. hmm I would maybe make that a bit more obvious and break

```
gRPCCredentialsOpt := WithGRPCDialOption(grpc.WithPerRPCCredentials(creds{cr}))
gRPCCredentialsOpt(c)
```

Reply:
@benhinchley done. Thanks.